### PR TITLE
[system-health] Make check interval more accurate

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
#### Why I did it

Healthd check system status every 60 seconds. However, running checker may take several seconds. Say checker takes X seconds, healthd takes (60 + X) seconds to finish one iteration. This implementation makes sonic-mgmt test case not so stable because the value X is hard to predict and different among different platforms. This PR introduces an interval 
compensation mechanism to healthd main loop.

#### How I did it

Introduces an interval compensation mechanism to healthd main loop: healthd should wait (60 - X) seconds for next iteration

#### How to verify it

Manual test 
Unit test

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211

### CODE REVIEW

**Review by keboliu — APPROVED** [review]

**Comment by prgeor** [comment]
@Junchao-Mellanox can you elaborate more on this `This implementation makes sonic-mgmt test case not so stable`

**Comment by Junchao-Mellanox** [comment]
> @Junchao-Mellanox can you elaborate more on this `This implementation makes sonic-mgmt test case not so stable`

Currently, sonic-mgmt has a test case like this:
1. mock something, e.g. PSU high temperature
2. wait 70 seconds 
3. check redis database and verify healthd put PSU high temperature error to STATE DB

The case failed intermittently because sometimes 70 seconds is not enough. If we introduce a time compensation to healthd main loop, healthd loop interval will be more accurate and 70 seoncds is enough. BTW, time compensation is a common way in sonic.

File: src/system-health/scripts/healthd [hunk 1]
```diff
@@ -79,18 +80,27 @@ class HealthDaemon(DaemonBase):
                 return
             sysmon = Sysmonitor()
             sysmon.task_run()
-            while 1:
-                stat = manager.check(chassis)
-                self._process_stat(chassis, manager.config, stat)
-
-                if self.stop_event.wait(manager.config.interval):
-                    break
+            while self._run_checker(manager, chassis):
```
Comment (prgeor) [inline src/system-health/scripts/healthd:83]: previously healthd was running continuously in while 1 loop. with this change the loop can break and healthd wont run. why this is changed?

File: src/system-health/scripts/healthd [hunk 2]
```diff
@@ -79,18 +80,27 @@ class HealthDaemon(DaemonBase):
...
+        begin = time.time()
+        stat = manager.check(chassis)
+        self._process_stat(chassis, manager.config, stat)
+        elapse = time.time() - begin
+        sleep_time_in_sec = manager.config.interval - elapse
+        if sleep_time_in_sec < 0:
+            self.log_notice(f'System health takes {elapse} seconds for one iteration')
+            sleep_time_in_sec = 1
+        if self.stop_event.wait(sleep_time_in_sec):
+            return False
```
Comment (prgeor) [inline src/system-health/scripts/healthd:101]: if return 'False' healthd will stop. can we just break?

File: src/system-health/scripts/healthd (hunk 1 above)
Comment (Junchao-Mellanox) [reply src/system-health/scripts/healthd:83]: Previous the loop still can be break, this logic does not change actually.
```
            while 1:
                stat = manager.check(chassis)
                self._process_stat(chassis, manager.config, stat)

                if self.stop_event.wait(manager.config.interval):
                    break
```

File: src/system-health/scripts/healthd (hunk 2 above)
Comment (Junchao-Mellanox) [reply src/system-health/scripts/healthd:101]: if stop_event is set, it means that healthd got a signal, and it should stop.

File: src/system-health/scripts/healthd (hunk 1 above)
Comment (Junchao-Mellanox) [reply src/system-health/scripts/healthd:83]: @prgeor

File: src/system-health/scripts/healthd (hunk 2 above)
Comment (Junchao-Mellanox) [reply src/system-health/scripts/healthd:101]: @prgeor

**Review by prgeor — APPROVED** [review]

**Comment by prgeor** [comment]
@liat-grozovik @lguohan please help merge

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**